### DISCUSSION TITLE
No clipboard in Workspace Prompts.

### DISCUSSION THREAD

**Discussion by ghost** [open-webui/open-webui#3871] [General]
# Bug Report

## Description

**Bug Summary:**
Can't integrate clipboard retrieval functionality in workspace prompts.

**Steps to Reproduce:**
Enter a workspace prompt like /spellcheck "Check spelling in the text: {{CLIPBOARD}}" (without quotes).

**Expected Behavior:**
When enter / and click on the above listed /spellcheck item, expect to see a prompt Like "Check spelling in the text: " + clipboard contents.

**Actual Behavior:**
Enter / and click on the above listed /spellcheck item, and the page hangs.

## Environment

- **Open WebUI Version:** [e.g., 0.1.120]
- ? (Pulled 3-days ago on 11th July. Can't find version info anywhere convenient?)
- **Ollama (if applicable):** [e.g., 0.1.30, 0.1.32-rc1]
- ollama version is 0.2.3
- 

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]
- Ubuntu 24.04
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]
- Firefox on the local Linux machine.
- Chrome on a distant Windows 11 machine.

## Reproduction Details

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by tjbck** [comment]
Unable to reproduce here, keep us updated if you find the solution!

**Reply by TFWol** [reply]
I have the same problem, except I have a console error instead of freezing.
![image](https://github.com/user-attachments/assets/17ef0089-b834-4a57-aaa8-1b826013f1e6)

It might have something to do with it not working if SSL isn't used like with what happened in [this project](https://github.com/josdejong/svelte-jsoneditor/issues/98). 

Over there [it was fixed](https://github.com/josdejong/svelte-jsoneditor/commit/d43a6465cd01f91fe8d1634038dfcf67fa578c81) and I noticed a similar workaround can used if you tie in [existing code](https://github.com/open-webui/open-webui/blob/ec99ac71214c4866381f6005627711e4d1f2e10f/src/lib/utils/index.ts#L240) into [the problematic code](https://github.com/open-webui/open-webui/blob/ec99ac71214c4866381f6005627711e4d1f2e10f/src/lib/components/chat/MessageInput/PromptCommands.svelte#L30)
